src/proposer/proposer.h

```diff
@@ -38,6 +42,12 @@ class Proposer {
 
   virtual bool IsStarted() = 0;
 
+  virtual bool GenerateBlock(CWallet *wallet,
```

Comment:
Wallet could be a reference.

Reply:
Suggest renaming to `TryPropose`.

Reply:
The wallet is now a reference, as for the renaming I answered to the correspondent comment on the `.cpp`.